rebac-admin-backend/v1/interfaces/identity_providers.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2024 Canonical Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+package interfaces
+
+import (
+	"context"
+
+	"github.com/canonical/identity-platform-admin-ui/rebac-admin-backend/v1/resources"
+)
+
+// IdentityProviderService defines an abstract backend to handle Roles related operations.
+type IdentityProviderService interface {
```

Comment:
It should be `...Providers...` (plural) due to consistency.

Reply:
Will fix